shell/platform/linux/fl_settings_portal.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_SETTINGS_PORTAL_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_SETTINGS_PORTAL_H_
+
+#include "flutter/shell/platform/linux/fl_settings.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlSettingsPortal,
+                     fl_settings_portal,
+                     FL,
+                     SETTINGS_PORTAL,
+                     GObject);
+
+/**
+ * FlSettingsPortal:
+ * #FlSettingsPortal reads settings from the XDG desktop portal.
+ */
+
+/**
+ * fl_settings_portal_new:
+ * @values: (nullable): a #GVariantDict with values for testing.
+ *
+ * Creates a new settings portal instance.
+ *
+ * Returns: a new #FlSettingsPortal.
+ */
+FlSettingsPortal* fl_settings_portal_new(GVariantDict* values);
```

Comment:
Since values is just for testing might be clearer to have `fl_settings_portal_new` and `fl_settings_portal_new_with_test_values`.